docs/en/ingest-management/security/logstash-certificates.asciidoc

```diff
@@ -1,7 +1,301 @@
 [[secure-logstash-connections]]
-= Configure SSL/TLS for {ls}
+= Configure SSL/TLS for the {ls} output
 
-coming[8.2.0]
+beta[]
 
 To send data from {agent} to {ls} securely, you need to configure Transport
-Layer Security (TLS).
+Layer Security (TLS). Using TLS ensures that your {agent}s send encrypted data
+to trusted {ls} servers, and that your {ls} servers receive data from trusted
+{agent} clients.
+
+[discrete]
+[[secure-logstash-prereqs]]
+== Prerequisites
+
+//REVIEWERS: Is there a specific subscription level required for this? The issue
+//mentions platinum, but as a general rule we point to the subsciptions page.
+//It's not clear to me which part of this feature requires plat.
+
+* Make sure your https://www.elastic.co/subscriptions[subscription level]
+supports output to {ls}.
+
+* On Windows, add port 8220 for {fleet-server} and 5044 for Logstash to the
+inbound port rules in Windows Advanced Firewall.
+
+* If you are connecting to a self-managed {es} cluster, you need the CA
+certificate that was used to sign the certificates for the HTTP layer of {es}
+cluster. For more information, refer to the
+{ref}/configuring-stack-security.html[{es} security docs].
+
+//REVIEWERS: I'm not going to explain the steps for securing the LS to ES
+//connection here because IMO that's beyond the scope of this doc and may
+//confused cloud users who don't need to know this.
+
+[discrete]
+[[generate-logstash-certs]]
+== Generate custom certificates and private keys
+
+You can use whatever process you typically use to generate PEM-formatted
+certificates. The examples shown here use the `certutil` tool provided by {es}.
+
+//REVIEWERS: The following worked for me, and I saw people asking on discuss. Is
+//there any reason we should not recommend this to cloud users? 
+
+TIP: The `certutil` tool is not available on {ecloud}, but you can still use it
+to generate certificates for {agent} to {ls} connections. Just
+https://www.elastic.co/downloads/elasticsearch[download an {es} package],
+extract it to a local directory, and run the `elasticsearch-certutil` command.
+There's no need to start {es}!
+
+//REVIEWERS: I did not set up /etc/hosts. Please help me refine the steps here.
+//for example, do I really need to set --ip and --dns? Or is there a better way?
+//Because I'm running all on the same machine (and using docker for the agent
+//deployment, I might be missing something.)
```

Comment:
I think if you're not setting up `/etc/hosts` then the `--dns` option is not necessary, but I may be wrong. The `--ip` is likely necessary if you're using Docker for the setup, since it'll need to be set for the Docker gateway. e.g. `192.168.65.2`. 

These options specify the valid host names and IP addresses for which the certificates generated will be considered valid.

Reply:
I'm going to leave this as-is. I think this is OK for beta at least.